x/staking/keeper/msg_server.go

```diff
@@ -226,15 +226,10 @@ func (k msgServer) Delegate(goCtx context.Context, msg *types.MsgDelegate) (*typ
 		if err := k.SafelyIncreaseTotalLiquidStakedTokens(ctx, tokens, false); err != nil {
 			return nil, err
 		}
-		if err := k.SafelyIncreaseValidatorLiquidShares(ctx, validator, shares); err != nil {
+		validator, err = k.SafelyIncreaseValidatorLiquidShares(ctx, valAddr, shares)
+		if err != nil {
 			return nil, err
 		}
-		// Note: this is required for downstream uses of the validator variable
-		// since the validator's liquid shares were updated above
-		validator, found = k.GetValidator(ctx, valAddr)
```

Comment:
we are sure about deleting this? I think we should keep this one

Reply:
I am having 1:1 to the upstream version that is in Gaia now

Reply:
my bad, all good, `SafelyIncreaseValidatorLiquidShares` now returns updated validator too

Reply:
`validator, err = k.SafelyIncreaseValidatorLiquidShares` updates it above, no?